### PR TITLE
Add placeholders for console support that explain how to do it.

### PR DESCRIPTION
**Pull request by reduz**
image

A large amount of users seem to infer it is not possible to publish games on consoles based on no export option available. Adding this placeholder should hopefully clear up this misunderstanding.

### CODE REVIEW

File: editor/project_export.cpp [hunk 1]
```diff
@@ -1301,6 +1318,15 @@ ProjectExportDialog::ProjectExportDialog() {
 	export_templates_error->add_child(download_templates);
 	download_templates->connect("pressed", callable_mp(this, &ProjectExportDialog::_open_export_template_manager));
 
+	export_to_console = memnew(LinkButton);
+	export_to_console->set_text(TTR("Learn more about exporting to consoles.."));
```
Comment (YeldhamDev) [inline editor/project_export.cpp:1322]: Should be 3 dots instead of 2 here.

Comment (reduz) [reply editor/project_export.cpp:1322]: ack, @akien-mga will kill me for this.

**Comment by MuffinManKen** [comment]
At the risk of being too nitpicky, this implies that getting the templates from a third-party is the only option.  If you are licensed you could certainly do it yourself.

File: editor/editor_export.cpp [hunk 2]
```diff
@@ -1408,6 +1408,37 @@ bool EditorExport::poll_export_platforms() {
 	return changed;
 }
 
+void EditorExport::add_export_console_placeholder_platforms() {
+	{
+		Ref<EditorExportConsolePlaceholder> ps;
+		ps.instance();
+		ps->console_name = "Sony PlayStation";
```
Comment (Calinou) [inline editor/editor_export.cpp:1415]: Should this be PlayStation or PlayStation 4? I'm not sure if we want to add a placeholder for each iteration of the console, especially as the PlayStation 5 will be released soon.

**Comment by DarkMessiah** [comment]
I think you need add message for XBox too, because UWP isn't real "console support" :)

File: editor/editor_export.cpp (hunk 2 above)
Comment (wkubiak) [reply editor/editor_export.cpp:1415]: It should be PlayStation 4, Xbox One and Nintendo Switch, as this is the current-gen.
Next-gen platform support placeholders should be added when those will be officially supported by anyone.

That being said, none of the listed Godot porting third-party companies currently provide support for PS4 so for transparency sake I don't think this platform should be mentioned anywhere at this time.

**Comment by wkubiak** [comment]
@MuffinManKen 
I agree. For the Nintendo Switch example, it should just state:
"Exporting to Nintendo Switch requires getting licensed with Nintendo. Learn more about exporting to consoles (link)."

@DarkMessiah
I agree that there should be Xbox One placeholder listed as well.

File: editor/editor_export.cpp (hunk 2 above)
Comment (reduz) [reply editor/editor_export.cpp:1415]: With 5 coming out, it makes no sense to put a number here IMO.

**Comment by Sslaxx** [comment]
This may be an obvious question, but I would assume (if you're licensed and all that jazz) Godot will not warn about export templates for Switch, etc. if they're present? If so, perhaps some kind of notice about subsequent QA and release procedures might be relevant instead.

**Comment by reduz** [comment]
Sorry, it seems the addition of the override keyword broke this PR so it needs to be redone. I won't have time to do this in the near future, so if anyone else wants to salvage it, feel free.

**Comment by Calinou** [comment]
Superseded by https://github.com/godotengine/godot/pull/42359.

**Pull request closed without merging**